kimchi/src/circuits/lookup/index.rs

```diff
@@ -411,6 +433,8 @@ impl<F: PrimeField + SquareRootField> LookupConstraintSystem<F> {
                     lookup_table8.push(eval);
                 }
 
+                // @volhovm: Do we need to enforce that there is at least one table
```

Comment:
We should actually ensure that *either* we have a table with ID 0 *or* that there was space for us to insert a 0 value.

Reply:
"space to insert zero value" is checking that the total number of lookup table rows is one less than `d1` domain minus zk rows?